R2API.Rules/RuleCatalogExtras.cs

```diff
@@ -0,0 +1,339 @@
+ */
+namespace R2API;
+
+/// <summary>
+/// A class for adding new RuleDefs and RuleCategories to the game's RuleCatalog
+/// </summary>
+#pragma warning disable CS0436 // Type conflicts with imported type
+[AutoVersion]
+#pragma warning restore CS0436 // Type conflicts with imported type
+public static partial class RuleCatalogExtras
+{
+    /// <summary>
+    /// The Rules submodule's plugin GUID
+    /// </summary>
+    public const string PluginGUID = R2API.PluginGUID + ".rules";
+    /// <summary>
+    /// The Rules submodule's plugin name
+    /// </summary>
+    public const string PluginName = R2API.PluginName + ".Rules";
+
+    /// <summary>
+    /// An index that represents the Difficulty RuleCategoryDef Index
+    /// </summary>
+    public const int DifficultyCategoryIndex = 0;
+
+    /// <summary>
+    /// An index that represents the Expansions RuleCategoryDef Index
+    /// </summary>
+    public const int ExpansionsCategoryIndex = 1;
+
+    /// <summary>
+    /// An index that represents the Artifacts RuleCategoryDef Index
+    /// </summary>
+    public const int ArtifactsCategoryIndex = 2;
+
+    /// <summary>
+    /// An index that represents the Items RuleCategoryDef index, hidden by default
+    /// </summary>
+    public const int ItemsCategoryIndex = 3;
+
+    /// <summary>
+    /// An index that represents the Equipments RuleCategoryDef index, hidden by default
+    /// </summary>
+    public const int EquipmentsCategoryIndex = 4;
+
+    /// <summary>
+    /// An index that represents the Misc RuleCategoryDef index, hidden by default
+    /// </summary>
+    public const int MiscCategoryIndex = 5;
+
+    private static bool _hooksEnabled = false;
+    private static Dictionary<int, List<RuleDef>> categoryToCustomRules = new Dictionary<int, List<RuleDef>>();
+    private static List<RuleCategoryDef> ruleCategoryDefs = new List<RuleCategoryDef>();
+
+    /// <summary>
+    /// Adds a new category to the RuleCatalog
+    /// </summary>
+    /// <param name="category">The category to add</param>
+    /// <returns>An index that represents this category's index in the catalog, -1 if the custom entries where already added.</returns>
+    public static int AddCategory(RuleCategoryDef category)
+    {
+        SetHooks();
+
+        if(RuleCatalog.availability.available)
+        {
+            return AddCategoryDirectlyToCatalog(category);
+        }
+
+        return AddCategoryInternal(category);
+
+        static int AddCategoryDirectlyToCatalog(RuleCategoryDef categoryDef)
+        {
+            RuleCatalog.allCategoryDefs.Add(categoryDef);
+            ruleCategoryDefs.Add(categoryDef);
+            return RuleCatalog.allCategoryDefs.Count - 1;
+        }
+
+        static int AddCategoryInternal(RuleCategoryDef categoryDef)
+        {
+            ruleCategoryDefs.Add(categoryDef);
+            return 5 + ruleCategoryDefs.Count;
```

Comment:
Magic number that would be better replaced with a variable or supplied with comment for future maintainers